test/game_connection/connector/test_cs_remote_connector.py

```diff
@@ -0,0 +1,216 @@
+from unittest.mock import AsyncMock, MagicMock
+
+import pytest
+from PySide6 import QtCore
+
+from randovania.game_connection.connector.cs_remote_connector import (
+    ITEM_RECEIVED_FLAG,
+    ITEM_SENT_FLAG,
+    CSRemoteConnector,
+)
+from randovania.game_connection.executor.cs_executor import CSExecutor, CSServerInfo, GameState, WeaponData
+from randovania.game_description.resources.inventory import Inventory, InventoryItem
+from randovania.games.game import RandovaniaGame
+from randovania.interface_common.players_configuration import INVALID_UUID
+
+
+@pytest.fixture(name="connector")
+def remote_connector():
+    executor_mock = MagicMock(CSExecutor)
+    executor_mock.server_info = MagicMock(CSServerInfo)
+    executor_mock.server_info.uuid = INVALID_UUID
+    executor_mock.server_info.platform = "Freeware"
+    connector = CSRemoteConnector(executor_mock)
+    return connector
+
+
+async def test_general_class_content(connector: CSRemoteConnector):
+    assert isinstance(connector.executor, MagicMock)
+
+    assert connector.game_enum == RandovaniaGame.CAVE_STORY
+    assert connector.description() == f"{RandovaniaGame.CAVE_STORY.long_name}: Freeware"
+
+    connector.Finished = MagicMock(QtCore.SignalInstance)
```

Comment:
You can connect the signal to a magic mock, then assert the mock was called

Reply:
do you have an example snippet?

Reply:
I thin of you look for `.connect` in the tests you'll find some uses. I'm pretty sure I did it somewhere